### PR TITLE
CMR-9250 Adding in CMR collection drafts.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: ingest-app/src/cmr/ingest/api/generic_documents.clj [hunk 1]
```diff
@@ -54,6 +51,20 @@
   [route-params]
   (common-concepts/singularize-concept-type (:concept-type route-params)))
 
+(def concept->spec-map
```
Comment (eudoroolivares2016) [inline ingest-app/src/cmr/ingest/api/generic_documents.clj:54]:
```suggestion
(def draft-concept->spec-map
```

File: search-app/src/cmr/search/services/parameters/parameter_validation.clj [hunk 2]
```diff
@@ -110,8 +110,7 @@
     :always-case-sensitive #{}
     :disallow-pattern #{}}))
 
-;; TODO: Generic work: multiple values need to be pulled from config files
```
Comment (eudoroolivares2016) [inline search-app/src/cmr/search/services/parameters/parameter_validation.clj:113]: Should we keep the TODO?

File: indexer-app/src/cmr/indexer/data/concepts/generic.clj [hunk 3]
```diff
@@ -64,11 +69,30 @@
     {(keyword field-name) value
      (keyword field-name-lower) value-lower}))
 
+(defn field->index
+  "Functions which convert a part of metadata to a name-value which can be added
+  to an index document. This defmulti is directed by looking for an :Indexer
```
Comment (eudoroolivares2016) [inline indexer-app/src/cmr/indexer/data/concepts/generic.clj:74]:
```suggestion
  to an index document. This defn is directed by looking for an :Indexer
```

**Review by eudoroolivares2016 — APPROVED** [review]

File: indexer-app/src/cmr/indexer/data/concepts/generic.clj (hunk 3 above)
Comment (eereiter) [reply indexer-app/src/cmr/indexer/data/concepts/generic.clj:74]: Fixed and replaced with function

File: ingest-app/src/cmr/ingest/api/generic_documents.clj (hunk 1 above)
Comment (eereiter) [reply ingest-app/src/cmr/ingest/api/generic_documents.clj:54]: fixed.

File: search-app/src/cmr/search/services/parameters/parameter_validation.clj (hunk 2 above)
Comment (eereiter) [reply search-app/src/cmr/search/services/parameters/parameter_validation.clj:113]: We do want to implement it, but we promised to get rid of these comments and write tickets a couple of PIs ago, So I am doing that when I see some leftovers.

File: common-lib/src/cmr/common/concepts.clj [hunk 4]
```diff
@@ -28,6 +28,18 @@
...
+  []
+  (vec
+   (filter #(not (string/includes? % "draft")) (get-generic-concept-types-array))))
+
+(defn get-draft-concept-types-array
+  "Gets all of the generic concept types that are drafts."
+  []
+  (vec
+   (filter #(string/includes? % "draft") (get-generic-concept-types-array))))
+
```
Comment (jaybarra) [inline common-lib/src/cmr/common/concepts.clj:42]: Since these don't take any input, these could just be `def`s

**Review by jaybarra — APPROVED** [review]

File: indexer-app/src/cmr/indexer/data/concepts/generic.clj [hunk 5]
```diff
@@ -110,8 +134,78 @@
...
+                             (let [fields (field->keyword-fields config)]
+                               (vec (concat total-fields fields))))
+                           []
+                           configs)]
+    (reduce (fn [data, key-name]
+              (let [value (get doc (keyword key-name))]
+                (into data (if (string? value)
+                             (vector value)
+                             value))))
+            (sequence nil)
```
[the remark this answers is not in the document]
Comment (eereiter) [reply indexer-app/src/cmr/indexer/data/concepts/generic.clj:161]: fixed.

File: indexer-app/src/cmr/indexer/data/concepts/generic.clj [hunk 6]
```diff
@@ -53,7 +39,26 @@
     {(keyword field-name) field-value
      (keyword field-name-lower) field-value-lower}))
 
-(defmethod field->index :default
+(defn field->index-simple-array-field
+  ;; The gets the a sub field of an array element and puts those values
+  ;; into a list so that each value can be searched on one of two indexes:
+  ;; the literal case, another is all lower case."
```
[the remark this answers is not in the document]
Comment (eereiter) [reply indexer-app/src/cmr/indexer/data/concepts/generic.clj:45]: Fixed

**Pull request merged**